sale_loyalty_order_line_link/models/sale_order.py

```diff
@@ -46,13 +26,16 @@ def _get_discounted_lines(self, program):
         """Hook method that allows to link lines from extra discount reward options"""
         # TODO: Should we add the lines that meet the criteria even if they aren't
         # on the filtered lines?
-        if program.discount_apply_on == "on_order":
+        if program.discount_applicability == "order":
             return self.order_line.filtered(lambda x: not x.is_reward_line)
-        elif program.discount_apply_on == "cheapest_product":
-            return self._get_cheapest_line()
-        elif program.discount_apply_on == "specific_products":
+        elif program.discount_applicability == "cheapest":
+            return self._cheapest_line()
+        elif program.discount_applicability == "specific":
             return self.order_line.filtered(
-                lambda x: x.product_id in program.discount_specific_product_ids
+                lambda x: x.product_id
+                in self.env["product.product"].search(
+                    program._get_discount_product_domain()
```

Comment:
Why don't you use the `_get_specific_discountable_lines` method (https://github.com/odoo/odoo/blob/adae60fc532cf6a853815f840cb49e93eb008cb7/addons/sale_loyalty/models/sale_order.py#L194)

Reply:
I was looking for the method to filter the lines by specific products and didn't realise that there was already a method that returned the lines for that type of reward. Thank you very much :)